### ISSUE TITLE
Issue with Angular 5.x

### ISSUE THREAD

**Issue by m-ghaoui** [surmon-china/ngx-quill-editor#31]
After upgrading my project to Angular 5, I now get this error when running `ng serve`:

```
ERROR in ./node_modules/ngx-quill-editor/quillEditor.component.ts
Module build failed: Error: C:\projects\event-system\event-system-admin\node_modules\ngx-quill-editor\quillEditor.component.ts is not part of the compilation output. Please check the other error messages for details.
    at AngularCompilerPlugin.getCompiledFile (C:\projects\event-system\event-system-admin\node_modules\@ngtools\webpack\src\angular_compiler_plugin.js:629:23)
    at plugin.done.then (C:\projects\event-system\event-system-admin\node_modules\@ngtools\webpack\src\loader.js:467:39)
    at <anonymous>
    at process._tickCallback (internal/process/next_tick.js:188:7)
 @ ./src/app/speaker/person-details/person-details.component.ts 18:0-103
 @ ./src/app/speaker/speaker.module.ts
 @ ./src/app/app.module.ts
 @ ./src/main.ts
 @ multi webpack-dev-server/client?http://0.0.0.0:0 ./src/main.ts
ERROR in ./node_modules/ngx-quill-editor/quillEditor.module.ts
Module build failed: Error: C:\projects\event-system\event-system-admin\node_modules\ngx-quill-editor\quillEditor.module.ts is not part of the compilation output. Please check the other error messages for details.
    at AngularCompilerPlugin.getCompiledFile (C:\projects\event-system\event-system-admin\node_modules\@ngtools\webpack\src\angular_compiler_plugin.js:629:23)
    at plugin.done.then (C:\projects\event-system\event-system-admin\node_modules\@ngtools\webpack\src\loader.js:467:39)
    at <anonymous>
    at process._tickCallback (internal/process/next_tick.js:188:7)
 @ ./src/app/speaker/speaker.module.ts 18:0-94
 @ ./src/app/app.module.ts
 @ ./src/main.ts
 @ multi webpack-dev-server/client?http://0.0.0.0:0 ./src/main.ts
```

There's a workaround for this, but it would be good if this issue is fixed.

**Comment by m-ghaoui** [comment]
The workaround is similar to the one mentioned here:

https://github.com/angular/angular/issues/20091#issuecomment-341376512

I managed to temporarily fix this issue by adding this to the `tsconfig.json` file:

```
  "include": [
    "src/**/*",
    "node_modules/ngx-quill-editor/quillEditor.component.ts",
    "node_modules/ngx-quill-editor/quillEditor.module.ts"
  ]
```

**Comment by micheleissa** [comment]
I'm trying what you mentioned and still getting the same error!

**Comment by m-ghaoui** [comment]
Hm. What does the error message say? Does it mention which file is 'is not part of the compilation output'?

**Comment by alan345** [comment]
Hi! I have almost same error. I have added to `tsconfig.json file:`

```
  "include": [
    "src/**/*",
    "node_modules/ngx-quill-editor/quillEditor.component.ts",
    "node_modules/ngx-quill-editor/quillEditor.module.ts"
  ]
```

and error is:
```
ERROR in ./node_modules/ngx-quill-editor/index.ts
Module build failed: Error: /Users/alan/app/arte/node_modules/ngx-quill-editor/index.ts is not part of the compilation output. Please check the other error messages for details.
    at AngularCompilerPlugin.getCompiledFile (/Users/alan/app/arte/node_modules/@ngtools/webpack/src/angular_compiler_plugin.js:629:23)
    at plugin.done.then (/Users/alan/app/arte/node_modules/@ngtools/webpack/src/loader.js:467:39)
    at <anonymous>
    at process._tickCallback (internal/process/next_tick.js:188:7)
 @ ./src/app/quote/quote.module.ts 34:0-53
 @ ./src/app/user/user.module.ts
 @ ./src/app/app.module.ts
 @ ./src/app/index.ts
 @ ./src/main.ts
 @ multi ./src/main.ts
```

Thanks!

**Comment by micheleissa** [comment]
The file it says the problem with is just a node_modules npm pckg that consists of a very simple component  .ts file along with some assets ( images and css )
Worth mentioning is that once I changed the version something else ( e.g. ~4.3.4 ) it worked.

**Comment by lamarh** [comment]
Apparently you cannot delvier non-compiled libraries with migration to Angular 5.x & CLI 1.5.0.  See https://github.com/angular/angular-cli/issues/8284

The ngx-quill-editor library will need to pre-compile its delivery.

**Comment by m-ghaoui** [comment]
Any ETA on the fix?

**Comment by stas-kh** [comment]
It would be great to have this problem fixed

**Comment by DanushkaAmith** [comment]
The same thing happen to me. It is working when I build the project with AOT but when I try to compile it same error popup like @alan345  Is there any solution?

**Comment by nnennajohn** [comment]
While you wait on this, for anyone else that runs into this problem, just grab the files and add to your own build directly like in the screenshot. 

I know it is not ideal... But deadlines have to be met. :) And as you can see from my screenshot, there are quite a few plugins i use that are yet to be updated and give the same exact error. 

Hope this helps someone. 

<img width="1231" alt="screenshot 2018-01-04 14 55 08" src="https://user-images.githubusercontent.com/1761197/34587985-8678303c-f15f-11e7-97cf-a6701b382c23.png">

**Comment by DanushkaAmith** [comment]
Hi Guys,
I found solution use this one [git repo](https://github.com/KillerCodeMonkey/ngx-quill)

**Comment by m-ghaoui** [comment]
@DanushkaAmith Thanks for  the tip! Sorry for the beginner question, but what is the best way to link to that repo using npm? I haven't quite understood how you can "make build install" an npm package directly off a git repo.

**Comment by SalathielGenese** [comment]
@m-ghaoui 

A github repo is located at `https://github.com/{ ORGNAME }/{ REPONAME }` where `ORGNAME` can actually be `USERNAME`.

`npm` allow you to `npm install ORGNAME/REPONAME` to get a package directly from github repo.

**Comment by m-ghaoui** [comment]
I've decided to migrate to https://github.com/Ledzz/angular2-tinymce

**Comment by Digihosp** [comment]
Include it in your tsconfig.app.json like this 
{
  "extends": "./tsconfig.json",
  "compilerOptions": {
    "outDir": "./out-tsc/app",
    "types": []
  },
  "include": [
    "src/**/*.ts",
    "node_modules/ngx-quill-editor"
  ],
  "exclude": [
    "src/test.ts",
    "src/**/*.spec.ts"
  ]
}

**Comment by mohanlsb** [comment]
install those package ```
"
"ngx-quill": "^8.0.0",
    "quill": "^1.3.7",

```"

in app.module.ts
import { QuillModule, QuillEditorComponent } from 'ngx-quill';
`QuillModule.forRoot({
      modules: {
        syntax: true
      }
    }),`

add providers as well 
in app.module.ts


providers: [QuillEditorComponent],

in HTML file 
`<quill-editor  [styles]="quillEditorStyle">`

in component.ts

quillEditorStyle = {
    height: '300px'
  };

![Screenshot 2020-02-11 at 10 16 15 PM](https://user-images.githubusercontent.com/41159056/74258354-301f4480-4d1c-11ea-93d0-b94c1ae2895d.png)


this work for me

hope you guyes like this
